### ISSUE TITLE
`Oban.pause_queue/2` and related functions always return `:ok` even if queues don't exist

### ISSUE THREAD

**Issue by vereis** [sorentwo/oban#553]
### Is your feature request related to a problem? Please describe.

The other day, we were working against a clone of one of our production databases in a staging environment. Because we handle a bunch of stuff via `Oban` that we absolutely _did not want to run_, we did an `Oban.pause_queue(queue: :default)` and saw it return `:ok`.

A few days later, we got reports and complaints, because it appears the queue was not in fact paused. Upon some investigation, this was likely due to `Oban.pause_queue(queue: :non_existing_queue_name)` always returning `:ok` making it susceptible to typos.

While we usually have `Oban.Pro` and `Oban.Web`, we didn't have the dashboard available in our staging environment 😢

### Describe the Solution You'd Like

I see you have a function in all related functions that validates `opts` passed in. I'm wondering if we could simply add the following check:

```elixir
defp validate_queue_opts!({:queue, queue}) do
  cond do
    (is_atom(queue) and not is_nil(queue)) or (is_binary(queue) and byte_size(queue) > 0) ->
          raise ArgumentError,
            "expected :queue to be a binary or atom (except `nil`), got: #{inspect(queue)}"

      # Either checking the provided config, or checking Oban.Registry
      is_nil(Registry.whereis(Oban, {:producer, to_string(queue)})) ->
          raise ArgumentError,
            "provided :queue does not exist, got: #{inspect(queue)}"

      true ->
        :ok
  end
end
```

### Describe Alternatives You've Considered

This could be a separate API; or perhaps, instead of returning `:ok` after queue manipulation, you can return a `queue_state()` instead? If so, given that `Oban.check_queue/2` crashes when provided an invalid queue, the same type of crash would happen following calling `Oban.pause_queue/2` (or related function) with an invalid queue name as well.

Perhaps there is a better way of doing this?

### Additional Context

N/A

**Comment by sorentwo** [comment]
There's a subtle but fundamental difference between `check_queue` and the other `_queue` functions. `check_queue` is only run locally, meaning you only get the queue's state on the local node. The other functions are either local or global. The command is executed through pub-sub in a "fire and forget" mode in the default global mode. Thus, affected queues/nodes don't respond to let the caller know if it worked or their state.

In a multi-node cluster it's possible that a couple of nodes are running queue `:a` and others are running queue `:b`, and we must be able to pause `:b` when making the call to nodes running `:a`.

It's possible to check whether a queue exists when `local_only` is set, but it isn't possible for global calls. Remember, nodes don't all have to run the same queues.

**Comment by vereis** [comment]
Ah okay, that makes sense -- that isn't a feature we're currently using so it didn't cross my mind! Thanks for letting me know.

Given that, do you think it's worth implementing such a check when `local_only` is set? I think I'd have appreciated the added safety, but at the same time I can understand validating the queue exists only when a certain arg is passed is both perhaps unexpected, _and_ potentially a little ugly...

**Comment by sorentwo** [comment]
I think the added safety is worth a little more validation there. With a little documentation, we can explain the difference, and with a helper function on `Registry` we can make the code look decent, too.

I'll leave this open and up for grabs if somebody wants to take a shot at it before I can.

**Comment by sorentwo** [comment]
Closed by #559
